src/t_set.c

```diff
@@ -798,8 +798,9 @@ void spopWithCountCommand(client *c) {
 
         /* todo: Move the spop notification to be executed after the command logic. */
 
-        /* Propagate this command as a DEL operation */
-        rewriteClientCommandVector(c,2,shared.del,c->argv[1]);
+        /* Propagate this command as a DEL or UNLINK operation */
+        robj *aux = server.lazyfree_lazy_server_del ? shared.unlink : shared.del;
+        rewriteClientCommandVector(c, 2, aux, c->argv[1]);
```

Comment:
This seems fine, it looks like we already do this in the expire command. I see we also do this in restore as well, so might as well also update that.

Reply:
yean, i didn't notice it, updated it.